lib/ProofSet.js

```diff
@@ -237,3 +309,52 @@ function _addToJSON(error) {
     writable: true
   });
 }
+
+async function _matchProofSet({
+  purposeToProofs, proofToSuite, purpose, proofSet, suites,
+  suiteMatchQueue, document, documentLoader, expansionMap
+}) {
+  for(const proof of proofSet) {
+    // first check if the proof matches the purpose; if it doesn't continue
+    if(!await purpose.match(proof, {document, documentLoader, expansionMap})) {
+      continue;
+    }
+
+    // next, find the suite that can verify the proof
+    let matched = false;
```

Comment:
It's not clear what happens if `matched` continues to be `false`? I'm guessing `matches` is processed in some way in the calling function... hard to follow this code.

Reply:
I'll add a comment about it.

Reply:
https://github.com/digitalbazaar/jsonld-signatures/pull/149/commits/bc539b6170398a2f2e7e791a9991b5648ef096ab